### ISSUE TITLE
Clean and organize GitHub issue and PR labels for this BSSW GitHub repo

### ISSUE THREAD

**Issue by bartlettroscoe** [betterscientificsoftware/betterscientificsoftware.github.io#374]
CC: @markcmiller86, @curfman, @bernhold, @rinkug,

Suggestions for list of labels:

* Focus:
  * BSSW content:
    * `content: blog` (current `blog`)
    * `content: curated` (current `curated content ideas`)
    * `content: event` (current `event`)
    * `content: article` (current `short article`)
    * `content: whatis` (current `whatis`)
    * `content: howto` (current `howto`)
  * `site internal` (current same)
* Type of issue (usually for `site internal` issues but could be for `content: ???` as well):
  * `type: enhancement` (current `enhancement`)
  * `type: bug` (current `bug`)
  * `type: help wanted` (current `help wanted`)
  * `type: question` (current `question`)
* Resolution type:
  * `resolution: duplicate` (current `duplicate`)
  * `resolution: wontdo` (current `wontfix`)
  * Default is that the issue was worked and completed

**Comment by bernhold** [comment]
I think we should eliminate `whatis` and `howto` and map them to the new `content: article`

Do we need more nuanced resolutions for ideas we decide not to develop?  Maybe not (or not until we gain more experience and observe a real value to having them).

**Comment by bartlettroscoe** [comment]
@bernhold said:

> I think we should eliminate `whatis` and `howto` and map them to the new `content: article`

The `content: <name>` labels should match whatever the decision on the contents types that comes out of #361.  But for now, can we just convert what the existing labels are as per the suggestions [above](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/374#issue-453148854)?  We can then update the labels `content: <name>` to match the new content types at the completion of #361.

Sound good?  Can I get some thumbs up?

**Comment by markcmiller86** [comment]
> I think we should eliminate `whatis` and `howto` and map them to the new `content: article`

I thought we decided (voted in fact) the name of  the content was to be **Original Article**

**Comment by jarrah42** [comment]
What about using "original" instead of "article" for the content type? Otherwise I'm ok with the suggested labels.

**Comment by bartlettroscoe** [comment]
I am okay with `content: original` or `content: original article`.  But let's make that decision in #361 and just implement that with the labels.

**Comment by markcmiller86** [comment]
First, I really think its important to be mindful of a) number of labels to choose from and b) ability of external visiters here to quickly understand the meaning of each label. So, we should include description text in the label definitions. We don't do that yet. And, that text should be just a short phrase or the labels page won't display it all.

Next, part of the reason for having labels, perhaps the only reason, is to facilitate quickly searching finding things. So, IMHO, every label we decide to have should be evaluated through the lens of *what does this help us do in searching*.

>     * `content: blog` (current `blog`)
>     * `content: curated` (current `curated content ideas`)
>     * `content: event` (current `event`)
>     * `content: article` (current `short article`)
>     * `content: whatis` (current `whatis`)
>     * `content: howto` (current `howto`)

I am not sure I see the value in prepending these with `content: `. It does have the helpful effect of keeping them sorted (together) in the drop down list when we are attaching them to an issue. Also, agree with @bernhold ... get rid of `content: whatis` and `content: howto`. I think it should be

- `event` (current `event`) (description: *an event or announcement of some date-specific activity*)
- `curated content` (current `curated content ideas`) (description: *typically one paragraph that describes/references another resource*)
- `original article` (current `short article`) (description: Content that is neither an *event* or *curated content*)

>   * `site internal` (current same)

I guess keep this though might just as well collapse into `admin` as with next suggestion...

> * Type of issue (usually for `site internal` issues but could be for `content: ???` as well):
>   
>   * `type: enhancement` (current `enhancement`)
>   * `type: bug` (current `bug`)
>   * `type: help wanted` (current `help wanted`)
>   * `type: question` (current `question`)

Do we really need these? I say collapse all these down to `admin` (description: *problems, questions, suggestions regarding site content or behavior*).

The article solicitation work **EB Members** routinely need to engage in is *help wanted* kind of activity. All that label might do is help would-be visiters to our repo potentially find content idea submissions here that we have yet to assign (or get a volunteer) author

> * Resolution type:
>   
>   * `resolution: duplicate` (current `duplicate`)
>   * `resolution: wontdo` (current `wontfix`)
>   * Default is that the issue was worked and completed

I don't think we need either of these. What does searching for `duplicate` do for us? Likewise for `wontdo`. If we *do* keep `wontfix`, we should keep that and not use `wontdo` as `wontfix` is GitHub convention whereas `wontdo` is not. In any case, just closing the issue with a comment such as *we decided not to move forward with this* is sufficient. We don't need a label. 

I like the intention of the `content: `, `type: `, etc. prepending but I think it just gets in the way. And, if we keep label set small, isn't really necessary.

Also, we have choice in coloring of these. I'd like to suggest admin-related be white or gray and content related be shades of a color such as green (or red) darker shades indicative of *size* of conent (e.g. event light green, curated middle-green, article dark-green).

**Comment by bartlettroscoe** [comment]
@markcmiller86 said:

> First, I really think its important to be mindful of a) number of labels to choose from and b) ability of external visiters here to quickly understand the meaning of each label. So, we should include description text in the label definitions. 

Yes, you do that with label comments which show as a pop-up when you move your mouse over them.  We did that for almost all of the labels in the Trilinos github repo.  It helps a lot IMHO.

> I am not sure I see the value in prepending these with `content: `. It does have the helpful effect of keeping them sorted (together) in the drop down list when we are attaching them to an issue. 

It does more than that.  It makes clear how these labels are related to each other.  If you have more than just a handful of labels, it really helps to prefix them.  For example, see the labels in:

* https://github.com/TriBITSPub/TriBITS/labels

That is not a lot of labels but it makes clear what these are.  Helps me pick them quickly.

We could get rid of the `type: enhancement` ... `type: question` labels, but as noted in "3. Nobody gets fired for opening an issue" in:

* https://ben.balter.com/2014/11/06/rules-of-communicating-at-github/

you want to use GitHub issues for as much communication as you can and these labels allow you to do that.

> "Resolution type:" ... I don't think we need either of these. 

I disagree.  If you close an issue for any reason other than that you actually completed what it says, then a label tells you why exactly it was closed.  Otherwise, you have to amend the title with "Duplicate: ???".  For example, if not for the "duplicate" label, just looking at the fact the the issue #128 is closed, you might have thought that was actually done (which was not the case when it was closed).  Otherwise, I am fine with `wontfix` instead of `wontdo` but it is strange to say that you "won't fix" and issue called "GitHub communication best practices".  What is there to fix or not fix?  That is software bug language.  This GitHub site does not have software.  But I guess you could still fix or not fix something about the internal utilities or something.

> I like the intention of the `content: `, `type: `, etc. prepending but I think it just gets in the way. And, if we keep label set small, isn't really necessary.

We can agree to disagree here.  See the discussion, for example, in trilinos/Trilinos#1619 and:

* https://medium.com/@dave_lunny/sane-github-labels-c5d2e6004b63
* https://jwong.co.uk/blog/2018/09/01/useful-github-features/

But this is a matter of opinion and there are pros and cons.  I guess we just need to put it up for a vote.

**Comment by markcmiller86** [comment]
> We can agree to disagree here

I think we'll have to. I just don't think a lot of what is needed here needs to be achieved via labels.

I agree on one thing though... `wontdo` is better than `wontfix` for reasons you describe.

**Comment by markcmiller86** [comment]
I guess I should amend my previous remark by saying, I often worry we're overthinking things on this project and can wind up adding complexity where there appears, initially at least, to be no need. I think its better to start simple and agree we'll make adjustments in process as the need arises. TriBits and Trilinos have tried and true practices that I am sure are relevant to many HPC/CSE projects. But, BSSw site repo seems a far cry simpler project and maybe doesn't require as heavy-weight processes. This at least is the explanation for the thinking cap I was wearing when considering our proposed labeling...as much simplification as possible.

**Comment by bartlettroscoe** [comment]
CC: @rinkug, @curfman, @davidbernholdt, @pagrubel

**Comment by rinkug** [comment]
We have moved to the label style "foo: foo". I have updated all the existing labels with this new format for now. We can reduce/simply labels, as needed, in the future. For now, no labels have been deleted so as to not disturb what labels have already been assigned to which issues

**Comment by bartlettroscoe** [comment]
@rinkug, the updated list of labels shown [here](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/labels) looks good to me.

Thanks for doing this!

**Comment by rinkug** [comment]
This issue is now being closed but we can reopen should a need arise in the future
